OpenRA.Mods.Common/Traits/BotModules/SendUnitToAttackBotModule.cs

```diff
@@ -0,0 +1,183 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Flags]
+	public enum AttackDistance
+	{
+		Closest = 0,
+		Furthest = 1,
+		Random = 2
+	}
+
+	[TraitLocation(SystemActors.Player)]
+	[Desc("Bot logic for units cannot cooperate with regular squad, like suicide unit with friendly damage or subterranean unit.")]
+	public class SendUnitToAttackBotModuleInfo : ConditionalTraitInfo
+	{
+		[Desc("Actors used for attack, and their base desire provided for attack desire.",
+			"When desire reach 100, AI will send them to attack.")]
+		public readonly Dictionary<string, int> ActorTypesAndAttackDesire = default;
+
+		[Desc("Target types that can be targeted.")]
+		public readonly BitSet<TargetableType> ValidTargets = new("Structure");
+
+		[Desc("Target types that can't be targeted.")]
+		public readonly BitSet<TargetableType> InvalidTargets;
+
+		[Desc("Should attack the furthest or closest target. Possible values are Closest, Furthest, Random")]
+		public readonly AttackDistance AttackDistance = AttackDistance.Closest;
+
+		[Desc("Attack order name.")]
+		public readonly string AttackOrderName = "Attack";
+
+		[Desc("Find target and try attack target in this interval.")]
+		public readonly int ScanTick = 463;
```

Comment:
Where is this value from. 😅

Reply:
it is a prime number to avoid excuting main function of Itick at the same time with other bot module.

Reply:
For example, if I set it to 500, and another bot module is 50, then we will have a same time excuting after 500 tick.

If we set it to a prime number like 463, the same time excuting will happen after 50*463.

Reply:
For example, if I set it to 500, and another bot module is 50, then we will have a same time excuting after 500 tick.

If we set it to a prime number like 463, the same time excuting will happen after 50*463.

Reply:
This shows that If we use prime number properly on these bot module, we will have better performance than use a rather large non-prime number.